### ISSUE TITLE
SSL: WRONG_VERSION_NUMBER

### ISSUE THREAD

**Issue by fessacchiotto** [Feuermurmel/youtube-podcast-gateway#12]
Hello!

There is an issue I found on Ubuntu ... Any ideas?

```
[65]: Connection was closed: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1977)
./run.sh: line 16: 19229 Segmentation fault      (core dumped) python . "${options[@]}"
```

**Comment by Feuermurmel** [comment]
Uh, oh, not running this on Ubuntu here, so I can't directly test it.

What Ubuntu release are you on and which Python version are you using? Also, which version(s) of libssl are installed?

**Comment by fessacchiotto** [comment]
Ubuntu 16.04.1 LTS

**Comment by fessacchiotto** [comment]
Libssl 1.0.0

**Comment by Feuermurmel** [comment]
Sadly I can'r reproduce your problem. I set up a Ubuntu 16.04.1 AMD64 VM, updated all packages and set up the Podcast Gateway.

I found a lot of reports of with this error message from a wide range of applications using _OpenSSL_ and _libssl_ with speculations that this could be a race-condition or other implementation problem of one of those libraries. The segmentation fault strongly hints at that.

The only thing I can suggest is wrapping the shell script with a small piece of magic:

```
while :; do ./run.sh; sleep 5; done
```

This will restart the server whenever it fails.
